bundles/org.eclipse.ui.monitoring/src/org/eclipse/ui/internal/monitoring/MonitoringStartup.java

```diff
@@ -17,21 +17,27 @@
 package org.eclipse.ui.internal.monitoring;
 
 import org.eclipse.core.runtime.Platform;
+import org.eclipse.e4.ui.workbench.UIEvents;
 import org.eclipse.jface.preference.IPreferenceStore;
 import org.eclipse.swt.widgets.Display;
-import org.eclipse.ui.IStartup;
 import org.eclipse.ui.PlatformUI;
 import org.eclipse.ui.internal.monitoring.preferences.MonitoringPreferenceListener;
 import org.eclipse.ui.monitoring.PreferenceConstants;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventHandler;
+import org.osgi.service.event.propertytypes.EventTopics;
 
 /**
  * Starts the event loop monitoring thread. Initializes preferences from {@link IPreferenceStore}.
  */
-public class MonitoringStartup implements IStartup {
+@Component(service = EventHandler.class)
+@EventTopics(value = UIEvents.UILifeCycle.APP_STARTUP_COMPLETE)
+public class MonitoringStartup implements EventHandler {
```

Comment:
You can even define a new composite annotation like this

```
@Component(service = EventHandler.class)
@EventTopics(value = UIEvents.UILifeCycle.APP_STARTUP_COMPLETE)
public @interface StartupComponent {

}
```

what the can be used like this 
```suggestion
@StartupComponent
public class MonitoringStartup implements EventHandler {
```

Reply:
If I try to add for example to the `UILifeCycle` class:
```
@Component(service = EventHandler.class)
@EventTopics(value = APP_STARTUP_COMPLETE)
public static @interface StartupCompleted {
}
```
I get the error `Invalid component implementation class 'StartupCompleted': annotation type.`

Reply:
Maybe `@Component` can't be placed on meta annotation, maybe its just a bug in PDE-DS ...